### DISCUSSION TITLE
How can i add an IPv4 or IPv6 adress ?

### DISCUSSION THREAD

**Discussion by caetano-zpp** [netbox-community/netbox#16883] [Help Wanted!]
I’m trying to add an IPv4 address to a VM inside a host. In the provided field, it does not find the addresses. I need to know where I can register these addresses to choose them during the VM registration 

![image](https://github.com/netbox-community/netbox/assets/175319441/0d3fae81-9371-4535-8a3e-d87632f20b05)

**Comment by MattDRogers** [comment]
Create an interface for your VM, then assign that interface an IP address. After that, edit your VM and you should be able to select the IP address from the drop menu.
It works the same as physical devices. IPs are assigned to interfaces. Hope that helps.

**Reply by candlerb** [reply]
There is also a custom script [here](https://github.com/netbox-community/customizations/blob/master/scripts/create_vm.py) that you can install, that creates a VM, interface and IP address all in one go.
